### PR TITLE
Refactor | logging for src ip

### PR DESCRIPTION
**Pull request by rushi47**
Fixes : #71 

Replace custom function with formatter from aaya itself.

### CODE REVIEW

File: dataplane/ebpf/src/ingress/udp.rs [hunk 1]
```diff
@@ -30,17 +30,16 @@ pub fn handle_udp_ingress(ctx: TcContext) -> Result<i32, i64> {
...
-        &ctx,
-        "Received a UDP packet destined for svc ip: {}.{}.{}.{} at port: {}",
-        daddr_dot_dec[0],
-        daddr_dot_dec[1],
-        daddr_dot_dec[2],
-        daddr_dot_dec[3],
-        u16::from_be(unsafe { (*udp_hdr).dest })
-    );
+    
+    match get_src_ip(&ctx) {
```
Comment (rushi47) [inline dataplane/ebpf/src/ingress/udp.rs:34]: @shaneutt  & @astoycos , not sure if this is idiomatic to use match here cause its still lot more line.
Any sugesstions ? or if this look I will refactor code from other part as well where old function was being

Comment (astoycos) [reply dataplane/ebpf/src/ingress/udp.rs:34]: so I _think_ you should just be able to do 

```
info!("Received a UDP packet destined for svc ip:: {:ipv4} at Port: {} ", unsafe { (*ip_hdr).daddr }, u16::from_be(unsafe {(*udp_hdr).dest}));
      
```

And avoid the `get_src_ip` function 
 
Also I think you can remove the `ip_from_int` as well

Comment (rushi47) [reply dataplane/ebpf/src/ingress/udp.rs:34]: Lol yeah this make sense no point in writing function

Comment (rushi47) [reply dataplane/ebpf/src/ingress/udp.rs]: I refactored this across udp, tcp and icmp. Also removed old function.
At same time i noticed while testing (could be another discussion altogether), I think tcp ep, doesnt route or not sure if its issue with my setup. I tried using main branch but same issue.

Comment (shaneutt) [reply dataplane/ebpf/src/ingress/udp.rs]: TCP is only partially implemented at this time and is not fully working yet, so don't worry. #15 is tracking this, but is incomplete.

As long as UDP is still working (the integration tests check this) and you've verified the IP address display is still good then we should be OK :+1:

File: dataplane/ebpf/src/egress/icmp.rs [hunk 2]
```diff
@@ -38,16 +38,8 @@ pub fn handle_icmp_egress(ctx: TcContext) -> Result<i32, i64> {
...
-    info!(
-        &ctx,
-        "Received a ICMP Unreachable packet destined for svc ip: {}.{}.{}.{}",
-        daddr_dot_dec[0],
-        daddr_dot_dec[1],
-        daddr_dot_dec[2],
-        daddr_dot_dec[3],
-    );
-
+    info!(&ctx, "Received a ICMP Unreachable packet destined for svc ip:: {:ipv4} ", u32::from_be(unsafe { (*ip_hdr).daddr }));
```
Comment (shaneutt) [inline dataplane/ebpf/src/egress/icmp.rs:41]:
```suggestion
    info!(&ctx, "Received a ICMP Unreachable packet destined for svc ip: {:ipv4} ", u32::from_be(unsafe { (*ip_hdr).daddr }));
```

File: dataplane/ebpf/src/ingress/tcp.rs [hunk 3]
```diff
@@ -28,16 +28,7 @@ pub fn handle_tcp_ingress(ctx: TcContext) -> Result<i32, i64> {
...
-    info!(
-        &ctx,
-        "Received a TCP packet destined for svc ip: {}.{}.{}.{} at port: {}",
-        daddr_dot_dec[0],
-        daddr_dot_dec[1],
-        daddr_dot_dec[2],
-        daddr_dot_dec[3],
-        u16::from_be(unsafe { (*tcp_hdr).dest })
-    );
+    info!(&ctx, "Received a TCP packet destined for svc ip:: {:ipv4} at Port: {} ", u32::from_be(unsafe { (*ip_hdr).daddr }), u16::from_be(unsafe { (*tcp_hdr).dest} ));
```
Comment (shaneutt) [inline dataplane/ebpf/src/ingress/tcp.rs:31]:
```suggestion
    info!(&ctx, "Received a TCP packet destined for svc ip: {:ipv4} at Port: {} ", u32::from_be(unsafe { (*ip_hdr).daddr }), u16::from_be(unsafe { (*tcp_hdr).dest} ));
```

File: dataplane/ebpf/src/ingress/udp.rs [hunk 4]
```diff
@@ -30,17 +30,8 @@ pub fn handle_udp_ingress(ctx: TcContext) -> Result<i32, i64> {
...
-        &ctx,
-        "Received a UDP packet destined for svc ip: {}.{}.{}.{} at port: {}",
-        daddr_dot_dec[0],
-        daddr_dot_dec[1],
-        daddr_dot_dec[2],
-        daddr_dot_dec[3],
-        u16::from_be(unsafe { (*udp_hdr).dest })
-    );
+    
+    info!(&ctx, "Received a UDP packet destined for svc ip:: {:ipv4} at Port: {} ", u32::from_be(unsafe { (*ip_hdr).daddr }), u16::from_be(unsafe { (*udp_hdr).dest} ));
```
Comment (shaneutt) [inline dataplane/ebpf/src/ingress/udp.rs:34]:
```suggestion
    info!(&ctx, "Received a UDP packet destined for svc ip: {:ipv4} at Port: {} ", u32::from_be(unsafe { (*ip_hdr).daddr }), u16::from_be(unsafe { (*udp_hdr).dest} ));
```

**Review by shaneutt — APPROVED** [review]
Couple small comments to resolve before we merge, but overall looks good thanks @rushi47!

Wish I had known that you could use `{:ipv4}` some time back, that's terribly convenient :joy:

File: dataplane/ebpf/src/ingress/udp.rs (hunk 1 above)
Comment (rushi47) [reply dataplane/ebpf/src/ingress/udp.rs]: Oh perfect thanks for clarifying @shaneutt

File: dataplane/ebpf/src/egress/icmp.rs (hunk 2 above)
Comment (rushi47) [reply dataplane/ebpf/src/egress/icmp.rs:41]: Damn sorry for this typo u had to fix shane. Thanks accepted it.

**Comment by rushi47** [comment]
Integration tests are failing because : 
Well interesting enough, same {:ipv4} formatter is behaving differently for icmp. Basically its loosing 2 values in last octect of ip address. Ex. instead of printing `172.0.2.100` its printing `172.0.2.1`. I am trying to figure out whats the issue.

**Comment by shaneutt** [comment]
> Integration tests are failing because : Well interesting enough, same {:ipv4} formatter is behaving differently for icmp. Basically its loosing 2 values in last octect of ip address. Ex. instead of printing `172.0.2.100` its printing `172.0.2.1`. I am trying to figure out whats the issue.

Sounds good thanks for looking into it @rushi47. If it ends up looking like there's something wrong on the `aya-log` side, would definitely encourage you to submit an issue there (https://github.com/aya-rs/aya/issues/new). They also have a discord server which is active and helpful as well. Let us know if you get stuck, and how we can help you!

**Comment by rushi47** [comment]
> > Integration tests are failing because : Well interesting enough, same {:ipv4} formatter is behaving differently for icmp. Basically its loosing 2 values in last octect of ip address. Ex. instead of printing `172.0.2.100` its printing `172.0.2.1`. I am trying to figure out whats the issue.
> 
> 
> 
> Sounds good thanks for looking into it @rushi47. If it ends up looking like there's something wrong on the `aya-log` side, would definitely encourage you to submit an issue there (https://github.com/aya-rs/aya/issues/new). They also have a discord server which is active and helpful as well. Let us know if you get stuck, and how we can help you!

Thank you shane, i was looking at code for aya log yesterday for formatter and tried to test it, with what they do internally it looked fine. But yes if this doesn't work I will raise issue.

**Review by shaneutt — DISMISSED** [review]
Ok thanks @rushi47

Seems reasonable to get this merged for now, appreciate the contribution!

**Comment by rushi47** [comment]
> One last thing, @rushi47 we are configured to required signed commits, if you could please rebase and sign your commits we can then merge. Here's a help doc if needed: https://docs.github.com/articles/about-gpg/

Done @shaneutt  I rebased and pushed again. Looks like our co-authered commits are unable to be verified. When I however it shows verified for me. Please lmk if anything else is needed.

**Review by shaneutt — APPROVED** [review]

**Pull request merged**